tools/infrasys/src/main.rs

```diff
@@ -0,0 +1,268 @@
+mod error;
+mod keys;
+mod root;
+mod s3;
+mod shared;
+
+use error::Result;
+use log::{error, info};
+use pubsys_config::InfraConfig;
+use sha2::{Digest, Sha512};
+use shared::KeyRole;
+use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger};
+use snafu::{ensure, OptionExt, ResultExt};
+use std::path::{Path, PathBuf};
+use std::{fs, process};
+use structopt::StructOpt;
+use tokio::runtime::Runtime;
+use url::Url;
+
+//   =^..^=   =^..^=   =^..^=  SUB-COMMAND STRUCTS  =^..^=   =^..^=   =^..^=
+
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+struct Args {
+    #[structopt(global = true, long, default_value = "INFO")]
+    log_level: LevelFilter,
+
+    // Path to Infra.toml  (NOTE: must be specified before subcommand)
+    #[structopt(long, parse(from_os_str))]
+    infra_config_path: PathBuf,
```

Comment:
We've seen positional arguments cause problems in the past. I wonder if it is better to use a regular argument such as `--infra-path`. My thinking is that if we wanted to add a subcommand that does not require Infra.toml, we would have a hard time maintaining compatibility with this positional global.

```suggestion
    #[structopt(global = true, long)]
    infra_path: PathBuf,
```

Reply:
I think infrasys would always need some sort of input file (which we might want to keep as Infra.toml?) that specifies what infra should be created (or what infra users already have). Currently pubsys uses the non-global infra_config_path, so for consistency we can keep it and change both when needed?

Reply:
We discussed this and the current name and positional nature of the argument matches pubsys. Resolving.